### PR TITLE
Use JsVal instead of JsRef in more places (nr. 5)

### PR DESCRIPTION
**Pull request by hoodmane**
Converts a lot more JsProxy methods to use JsVals.

### CODE REVIEW

File: src/core/jsproxy.c [hunk 1]
```diff
@@ -1239,15 +1224,90 @@ static PyMethodDef JsProxy_object_values_MethodDef = {
...
+  if (result >= 0) {
+    return result;
+  }
+  // Something went wrong. Case work:
+  // * -1: Either `val.size` or `val.length` was a getter which managed to raise
+  //    an error. Rude. (Also we don't defend against this in hiwire_has_length)
+  // * -2: Doesn't have a length or size, or they aren't of type "number".
+  //   But `hiwire_has_length` returned true? So it must have changed somehow.
+  // * -3: Length was >= 2^{31}
+  // * -4: Length was negative
```
Comment (ryanking13) [inline src/core/jsproxy.c:1273]: I think the comment of `-3` and `-4` is reversed.

**Review by ryanking13 — APPROVED** [review]
Thanks, @hoodmane!

File: src/core/jsproxy.c (hunk 1 above)
Comment (hoodmane) [reply src/core/jsproxy.c:1273]: This is probably a good place to use `#define`s.

**Comment by hoodmane** [comment]
Thanks for the review @ryanking13!

**Pull request merged**